clients/client-java/src/main/java/com/datastrato/gravitino/client/GravitinoCatalog.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.client;
+
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.Namespace;
+import com.datastrato.gravitino.dto.AuditDTO;
+import com.datastrato.gravitino.dto.CatalogDTO;
+import com.datastrato.gravitino.dto.requests.SchemaCreateRequest;
+import com.datastrato.gravitino.dto.requests.SchemaUpdateRequest;
+import com.datastrato.gravitino.dto.requests.SchemaUpdatesRequest;
+import com.datastrato.gravitino.dto.responses.DropResponse;
+import com.datastrato.gravitino.dto.responses.EntityListResponse;
+import com.datastrato.gravitino.dto.responses.SchemaResponse;
+import com.datastrato.gravitino.exceptions.NoSuchCatalogException;
+import com.datastrato.gravitino.exceptions.NoSuchSchemaException;
+import com.datastrato.gravitino.exceptions.NonEmptySchemaException;
+import com.datastrato.gravitino.exceptions.SchemaAlreadyExistsException;
+import com.datastrato.gravitino.rel.Schema;
+import com.datastrato.gravitino.rel.SchemaChange;
+import com.datastrato.gravitino.rel.SupportsSchemas;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Gravitino Catalog is the base class for all the catalog. It provides the common methods for
+ * managing schemas in a catalog. With {@link GravitinoCatalog}, users can list, create, load, alter
+ * and drop a schema with specified identifier.
+ */
+public class GravitinoCatalog extends CatalogDTO implements SupportsSchemas {
```

Comment:
I was thinking of making this as an abstract class with the name "BaseSchemaCatalog"